components/data-provider/src/structs/dates.rs

```diff
@@ -0,0 +1,124 @@
+// Date
+#[cfg(feature = "invariant")]
+use crate::prelude::*;
+
+/// Gets a locale-invariant default struct given a data key in this module's category.
+#[cfg(feature = "invariant")]
+pub(crate) fn get_invariant(data_key: &DataKey) -> Option<DataResponse<'static>> {
+    use crate::invariant::make_inv_response;
+    if data_key.category != DataCategory::Dates {
+        return None;
+    }
+    match (data_key.sub_category.as_str(), data_key.version) {
+        ("gregory", 1) => make_inv_response::<gregory::DatesV1>(),
+        _ => None,
+    }
+}
+
+pub mod gregory {
+    use serde::{Deserialize, Serialize};
+    use std::borrow::Cow;
+    #[derive(Debug, PartialEq, Clone, Deserialize, Serialize, Default)]
+    pub struct DatesV1 {
+        pub symbols: DateSymbolsV1,
+
+        pub patterns: PatternsV1,
+    }
+
+    #[derive(Debug, PartialEq, Clone, Deserialize, Serialize, Default)]
+    pub struct DateSymbolsV1 {
+        pub months: months::ContextsV1,
+
+        pub weekdays: weekdays::ContextsV1,
+
+        pub day_periods: day_periods::ContextsV1,
+    }
+
+    #[derive(Debug, PartialEq, Clone, Deserialize, Serialize, Default)]
+    pub struct PatternsV1 {
+        pub date: patterns::StylePatternsV1,
+
+        pub time: patterns::StylePatternsV1,
+
+        pub date_time: patterns::StylePatternsV1,
+    }
+
+    macro_rules! symbols {
+        ($name: ident, $expr: ty) => {
+            pub mod $name {
+                use super::*;
+
+                #[derive(Debug, PartialEq, Clone, Deserialize, Serialize)]
+                pub struct SymbolsV1(pub $expr);
+
+                symbols!();
+            }
+        };
+        ($name: ident, $($element: ident: $ty: ty),*) => {
+            pub mod $name {
+                use super::*;
+
+                #[derive(Debug, PartialEq, Clone, Deserialize, Serialize)]
+                pub struct SymbolsV1 {
+                    $(pub $element: $ty),*
+                }
+                symbols!();
+            }
+        };
+        () => {
+            #[derive(Debug, PartialEq, Clone, Deserialize, Serialize, Default)]
+            pub struct WidthsV1 {
+                #[serde(skip_serializing_if = "Option::is_none")]
+                pub abbreviated: Option<SymbolsV1>,
+
+                #[serde(skip_serializing_if = "Option::is_none")]
+                pub narrow: Option<SymbolsV1>,
+
+                #[serde(skip_serializing_if = "Option::is_none")]
+                pub short: Option<SymbolsV1>,
+
+                #[serde(skip_serializing_if = "Option::is_none")]
+                pub wide: Option<SymbolsV1>,
+            }
+
+            #[derive(Debug, PartialEq, Clone, Deserialize, Serialize, Default)]
+            pub struct ContextsV1 {
```

Comment:
You can probably flatten this. `stand_alone` is for Intl.DisplayNames. I think that for Intl.DateTimeFormat, you should use `format` by itself in general.

Reply:
Would you then recommend that when we add `DisplayNames` we keep `stand_alone` in a separate key instead of `Dates`?

Reply:
Yeah, I think the display names belong in their own data key.  CLDR puts them together because that's what is most relevant to translators, but for ICU4X API, they are in different components, which is fine.

Reply:
I verified with @sffc on Slack and confirmed that it's ok to keep `stand_alone` in `dates` key for now. We can discuss the adjustments to how the keys are divided later, but my main argument is that `DateTimeFormat` will need both `format` and `stand_alone` contexts.